### DISCUSSION TITLE
SwitchAction: Ich bekomme beim Click auf das Element kein event

### DISCUSSION THREAD

**Discussion by mcuiobroker** [Zefau/ioBroker.jarvis#2163] [Q&A]
Kann man den click-event von SwitchAction und co. "durchschleifen"?
Wenn ich auf das Element klicke bekomme ich bei den Elementen kein event.

**Comment by Zefau** [comment]
Mal angenommen ich könnte es "durchschleifen", wie würdest du es abfangen?

**Comment by mcuiobroker** [comment]
Fange es normalerweise immer so ab:
```
document.addEventListener("click", (event) => {
		if (isConfig()){
		}else{
			setTimeout(function(){
				activate9HKTClicks();
			},1000);
		}	
	});
```

**Reply by Zefau** [reply]
Damit fängst du einen Klick auf `document` (also die Seite) ab. Wenn du `document.getElementById( ... hier die ID ... )` oder `getElementsByClassName( ... hier die Klasse ..)` (Achtung: Hier im Ergebnis ein Array, da potential mehrere Elemente) machst, dann kannst du dieselbe Logik auch auf Elemente anwenden.
